src/GitHub.VisualStudio/Services/UsageTracker.cs

```diff
@@ -184,9 +163,24 @@ void UpdateTimer(bool initialCall)
             // Subsequent timer ticks should occur every 8 hours.
             timer.Interval = TimeSpan.FromHours(8);
 
+            if (userSettings == null)
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
I'm glad to see you're taking care to switch to the main thread when necessary, and using this mechanism in particular.